libraries/WiFi/src/WiFiClient.cpp

```diff
@@ -219,22 +224,33 @@ int WiFiClient::connect(IPAddress ip, uint16_t port)
 {
     return connect(ip,port,_timeout);
 }
+
 int WiFiClient::connect(IPAddress ip, uint16_t port, int32_t timeout_ms)
 {
+    struct sockaddr_storage serveraddr = {};
     _timeout = timeout_ms;
-    int sockfd = socket(AF_INET, SOCK_STREAM, 0);
+    int sockfd = -1;
+
+    if (ip.type() == IPv6) {
```

Comment:
Does this need a check of WIFI_WANT_IP6_BIT ... not sure if it is valid to try to connect to IPv6 if you don't have an IPv6 address. (But then, why would they even be passing it in). Probably okay to just let the connect fail.

Reply:
yeah... still brainstorming on that whole matrix of what you want, what you have and what can you actually get. Open to suggestions